### PR TITLE
[AF-2366] BC Maven Archetype Support

### PR DESCRIPTION
**Pull request by caponetto**
**Task**: [AF-2366](https://issues.redhat.com/browse/AF-2366)

**Related PRs**:
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1151
- https://github.com/kiegroup/kie-soup/pull/124
- https://github.com/kiegroup/appformer/pull/872
- https://github.com/kiegroup/kie-wb-common/pull/3098
- https://github.com/kiegroup/jbpm-wb/pull/1406
- https://github.com/kiegroup/drools-wb/pull/1278
- https://github.com/kiegroup/optaplanner-wb/pull/352
- https://github.com/kiegroup/kie-wb-distributions/pull/1005

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-archetype-mgmt/kie-wb-common-archetype-mgmt-api/src/main/java/org/kie/workbench/common/screens/archetype/mgmt/shared/events/ArchetypeListUpdatedEvent.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Objects;
+
+import org.jboss.errai.common.client.api.annotations.MapsTo;
+import org.jboss.errai.common.client.api.annotations.Portable;
+import org.kie.workbench.common.screens.archetype.mgmt.shared.model.ArchetypeListOperation;
+
+import static org.kie.soup.commons.validation.PortablePreconditions.checkNotNull;
+
+@Portable
+public class ArchetypeListUpdatedEvent {
```
[the remark this answers is not in the document]
Comment (caponetto) [reply kie-wb-common-screens/kie-wb-common-archetype-mgmt/kie-wb-common-archetype-mgmt-api/src/main/java/org/kie/workbench/common/screens/archetype/mgmt/shared/events/ArchetypeListUpdatedEvent.java]: Thanks, @paulovmr! Please check my last commit.

**Comment by caponetto** [comment]
jenkins please retest this

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java [hunk 2]
```diff
@@ -343,6 +344,22 @@ private void addDataSetPerspective() {
         }
     }
 
+    private void addArchetypesPerspective() {
+        if (hasAccessToPerspective(PerspectiveIds.ARCHETYPE_MANAGEMENT)) {
+            adminPage.addTool("root",
+                              constants.Archetypes(),
+                              new Sets.Builder().add("fa").add("fa-file-code-o").build(),
+                              "advanced",
```
Comment (sthundat) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java]: Define a constant for the literal "advanced".It is using 3 times in the file

Comment (caponetto) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/admin/DefaultAdminPageHelper.java]: Changing code where I originally haven't touched could lead to unwanted conflicts, but I've applied the changes you've requested. Please check my last commit.

**Comment by caponetto** [comment]
Hi @sthundat, are you still reviewing? This is the only PR you haven't approved.

**Pull request merged**